pubsub.go

```diff
@@ -25,34 +25,21 @@ func timeoutOk(c Conn) Conn {
 }
 
 func (toc *timeoutOkConn) Decode(u resp.Unmarshaler) error {
-	err := toc.Conn.Decode(timeoutOkUnmarshaler{u})
+	err := toc.Conn.Decode(u)
 	if err == nil {
 		return nil
 	}
-	if te, ok := err.(errTimeout); ok {
-		err = te.err
-	} else {
+	if !isTemporaryTimeout(err) {
 		toc.Close()
 	}
 	return err
 }
 
-type timeoutOkUnmarshaler struct {
```

Comment:
In thinking about this more I've realized that timeoutOk isn't needed at all, but I'll deal with that after this PR so I don't subject you to more than I have to :P